microsoft-edge/webview2/concepts/windowed-vs-visual-hosting.md

```diff
@@ -1,436 +1,194 @@
-
-If your WebView2 app uses visual hosting:
+## Visual hosting: For more granular control over layout
 
-* Inputs are routed to the app's `HWND` and must be configured to send the spatial input (for example, mouse, touch, and pen) based on positions, _not_ what currently has focus, such as a keyboard.
-
-The following APIs can be used when configuring WebView2 in a visual hosting environment:
+When using _Visual hosting_, your host app receives spatial input (such as mouse or touch input) from the user, and forwards this input to the WebView2 control.  Visual hosting requires the app to do the same window management as Windowed hosting, but has additional window management requirements regarding:
+* Scaling the contents.
+* Routing spatial inputs (such as mouse, touch, or pen).
 
 
 <!-- ------------------------------ -->
-#### Composition-based rendering
-
-For composition-based WebView2 rendering, use the `CoreWebView2Environment` to create a `CoreWebView2CompositionController`. The `CoreWebView2CompositionController` implements all the APIs as `CoreWebView2Controller`, but includes additional APIs specific to composition-based rendering.
-
-##### [.NET/C#](#tab/dotnetcsharp)
-
-* [CoreWebView2CompositionController Class](/dotnet/api/microsoft.web.webview2.core.corewebview2compositioncontroller)
-* [CoreWebView2Environment.CreateCoreWebView2CompositionControllerAsync Method](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.createcorewebview2compositioncontrollerasync)
+#### Requirements for scaling the contents
 
-##### [WinRT/C#](#tab/winrtcsharp)
+Per composition-based rendering, a WebView2 control is part of a Visual tree, so by default, it's rendered at the scale of its ancestor Visual.  For example, if a WebView2's ancestor Visual is scaled 2x, then the WebView2's contents are also rendered at 2x scale.  But because the WebView2 is not scaling its own contents, they're blurry.
 
-* [CoreWebView2CompositionController Class](/dotnet/api/microsoft.web.webview2.core.corewebview2compositioncontroller)
-* [CoreWebView2Environment.CreateCoreWebView2CompositionControllerAsync Method](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.createcorewebview2compositioncontrollerasync)
-
-##### [Win32/C++](#tab/win32cpp)
-
-* [ICoreWebView2CompositionController interface](/microsoft-edge/webview2/reference/win32/icorewebview2compositioncontroller)
-* [ICoreWebView2Environment::CreateCoreWebView2CompositionController method](/microsoft-edge/webview2/reference/win32/icorewebview2environment3#createcorewebview2compositioncontroller)
-
----
+To resolve this, the app can disable default Visual scaling on the WebView2, and instead use the Rasterization Scale APIs to apply the intended Visual scaling.  This results in the WebView2's contents rendering at the correct scale.
```

Comment:
Upon reconsideration I think 'undo' would be better than 'disable'

Reply:
result:
```
To resolve this, the app can undo the default Visual scaling of the WebView2, 
and instead use the Rasterization Scale APIs to apply the intended Visual scaling.
```